tests/versioning_3_test.go

```diff
@@ -25,39 +25,579 @@
 package tests
 
 import (
+	"context"
+	"fmt"
+	"sync/atomic"
+	"testing"
 	"time"
 
+	"github.com/dgryski/go-farm"
 	"github.com/pborman/uuid"
+	"github.com/stretchr/testify/suite"
+	commandpb "go.temporal.io/api/command/v1"
 	commonpb "go.temporal.io/api/common/v1"
+	deploymentpb "go.temporal.io/api/deployment/v1"
 	enumspb "go.temporal.io/api/enums/v1"
 	taskqueuepb "go.temporal.io/api/taskqueue/v1"
+	"go.temporal.io/api/workflow/v1"
 	"go.temporal.io/api/workflowservice/v1"
+	deploymentspb "go.temporal.io/server/api/deployment/v1"
+	"go.temporal.io/server/api/historyservice/v1"
+	"go.temporal.io/server/api/matchingservice/v1"
+	"go.temporal.io/server/common"
+	"go.temporal.io/server/common/dynamicconfig"
+	"go.temporal.io/server/common/payloads"
+	"go.temporal.io/server/common/primitives/timestamp"
+	"go.temporal.io/server/common/testing/taskpoller"
+	"go.temporal.io/server/common/testing/testvars"
+	"go.temporal.io/server/common/tqid"
 	"go.temporal.io/server/tests/testcore"
 	"google.golang.org/protobuf/types/known/durationpb"
 )
 
-func (s *VersioningIntegSuite) TestPinnedWorkflow() {
-	// TODO (shahab) implement this with TaskPoller etc.
-	s.startWorkflow()
+var (
+	tqTypeWf         = enumspb.TASK_QUEUE_TYPE_WORKFLOW
+	tqTypeAct        = enumspb.TASK_QUEUE_TYPE_ACTIVITY
+	unspecified      = enumspb.VERSIONING_BEHAVIOR_UNSPECIFIED
+	pinned           = enumspb.VERSIONING_BEHAVIOR_PINNED
+	unpinned         = enumspb.VERSIONING_BEHAVIOR_AUTO_UPGRADE
```

Comment:
these are package-level names, and the package is shared with all functional tests. I don't think we should be using such generic names